### PR TITLE
removed soft delete from CartItem model; added Order and OrderItem mo…

### PR DESCRIPTION
**Pull request by YuraKhilko**
…dels with liquibase changes: orders and order_items tables creation; added corresponding repository; added OrderService; added OrderController with corresponding DTOs;

### CODE REVIEW

File: src/main/java/bookstore/model/OrderItem.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "order_items")
+public class OrderItem {
```
Comment (Rommelua) [inline src/main/java/bookstore/model/OrderItem.java:20]: We have  Set<OrderItem> - add equals/hashCode methods

File: src/main/java/bookstore/repository/OrderRepository.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import bookstore.model.Order;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface OrderRepository extends JpaRepository<Order, Long> {
+    @EntityGraph(attributePaths = "orderItems")
+    List<Order> findAllByUser_Id(Long userId);
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/OrderRepository.java]:
```suggestion
    List<Order> findAllByUserId(Long userId);
```

File: src/main/java/bookstore/repository/OrderRepository.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    @EntityGraph(attributePaths = "orderItems")
+    Optional<Order> findByIdAndUser_Id(Long orderId, Long userId);
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/OrderRepository.java]:
```suggestion
    Optional<Order> findByIdAndUserId(Long orderId, Long userId);
```

File: src/main/java/bookstore/repository/OrderRepository.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Query("FROM Order o "
+            + "INNER JOIN FETCH o.orderItems oi "
+            + "INNER JOIN FETCH o.user u "
+            + "WHERE o.id = :orderId "
+            + "AND u.id = :userId "
+            + "AND oi.id = :itemId")
+    Optional<Order> findByIdAndUser_IdAAndOrderItem_Id(Long orderId, Long userId, Long itemId);
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/OrderRepository.java]: You use it for one OrderItem
better add to OrderItemRepository
Optional<OrderItem> findByIdAndOrderId(Long id, Long orderId)

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/bookstore/model/OrderItem.java (hunk 1 above)
Comment (YuraKhilko) [reply src/main/java/bookstore/model/OrderItem.java:20]: methods have been added and also to Role, CartItem and Book.

File: src/main/java/bookstore/dto/order/UpdateOrderDto.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package bookstore.dto.order;
+
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+
+@Data
+public class UpdateOrderDto {
+    @NotBlank
+    private String status;
```
Comment (aaananas) [inline src/main/java/bookstore/dto/order/UpdateOrderDto.java]: you can easily accept a enum here, no need to parse it later

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/bookstore/dto/order/UpdateOrderDto.java (hunk 5 above)
Comment (YuraKhilko) [reply src/main/java/bookstore/dto/order/UpdateOrderDto.java]: thank you

**Review by aaananas — APPROVED** [review]

**Pull request merged**